### PR TITLE
Handle -1 in --shapeInformation option and model signature

### LINKED ISSUE #1995: Unable to compile model with -1 in --shapeInformation
After the latest llvm update, using `--shapeInformation` to make a static dimension static causes a compile failure.

For example compiling [yolov3-12.onnx](https://github.com/onnx/models/blob/main/vision/object_detection_segmentation/yolov3/model/yolov3-12.onnx) input0 normally has shape [-1,3,-1,-1]. To make the image shape static but keep the batch size dynamic with:
```
--O3 --EmitLib yolov3-12.onnx --onnx-op-stats TXT --shapeInformation 0:-1x3x416x416
```
now gives failure:
```
error: invalid tensor dimension size
onnx-mlir: /build/llvm-project/mlir/include/mlir/IR/StorageUniquerSupport.h:153: static ConcreteT mlir::detail::StorageUserBase<ConcreteT, BaseT, StorageT, UniquerT, Traits>::get(mlir::MLIRContext*, Args ...) [with Args = {llvm::ArrayRef<long int>, mlir::Type, mlir::Attribute}; ConcreteT = mlir::RankedTensorType; BaseT = mlir::TensorType; StorageT = mlir::detail::RankedTensorTypeStorage; UniquerT = mlir::detail::TypeUniquer; Traits = {mlir::SubElementTypeInterface::Trait, mlir::ShapedType::Trait}]: Assertion `succeeded(ConcreteT::verify(getDefaultDiagnosticEmitFn(ctx), args...))' failed.
```

### PR DESCRIPTION
**Pull request by tungld**
After updating LLVM, we internally use `ShapedType::kDynamic`(that is `std::numeric_limits<int64_t>::min()`) because that is the way MLIR handles dynamic dimension.

But, for onnx-mlir's users, it is convenient to use `-1` to denote a dynamic dimension, e.g. in the model signatures or `--shapeInformation` option, which is easy to remember.

This patch fixes #1995 #1996 issues and add a LIT test for lowering `onnx.EntryPoint` to `krnl.entry_point`.

Signed-off-by: Tung D. Le <tung@jp.ibm.com>

### CODE REVIEW

**Review by AlexandreEichenberger — APPROVED** [review]
LGTM pending the addition of one assert, listed below.

File: src/Builder/ModelInputShaper.hpp
Comment (tungld) [reply src/Builder/ModelInputShaper.hpp:60]: `kDynamic` from MLIR is no longer `-1`, it is now `std::numeric_limits<int64_t>::min()`. `-1` is used for onnx-mlir's users only when they want to denote a dynamic dim in `--shapeInformation` option or model signature.

Comment (AlexandreEichenberger) [reply src/Builder/ModelInputShaper.hpp:60]: Got it, so it's good to go then. Maybe change the name to `kDynamicForPrinting` so that it is clearer.

**Comment by cjvolzka** [comment]
I gave this test spin today and can confirm it will fix both https://github.com/onnx/onnx-mlir/issues/1995 and https://github.com/onnx/onnx-mlir/issues/1996 once it merges.

**Pull request merged**
